walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -60,50 +52,74 @@
   }
 
   /** 
-   * Returns an ArrayList of all busy TimeRanges.
+   * Returns an ImmutableList of all busy TimeRanges.
    * Check all events for the day and add TimeRanges of events with our mandatory attendees.
   */
-  private ArrayList<TimeRange> findBusyRanges(Collection<String> attendees, Collection<Event> events) {
-    ArrayList<TimeRange> busyRanges = new ArrayList<TimeRange>();
-    for (Event event : events){
-      for (String attendee : attendees) {
-        if (event.getAttendees().contains(attendee)){
-          busyRanges.add(event.getWhen());
-        }      
-      }
+  private ImmutableList<TimeRange> findBusyRanges(Collection<String> attendees, Collection<Event> events) {
+    ArrayList<TimeRange> busyRanges = new ArrayList<>();
+    for (Event event : events) {
+        // Checks if we have any overlap of attendees
+        if (!Collections.disjoint(attendees, event.getAttendees())) {
+            busyRanges.add(event.getWhen());
+        }
     }
-    return busyRanges;
+
+    Collections.sort(busyRanges, TimeRange.ORDER_BY_START);
+
+    return ImmutableList.<TimeRange>builder() 
+              .addAll(busyRanges) 
+              .build();
```

Comment:
You can do all of this with ImmutableList.sortedCopyOf()

https://guava.dev/releases/23.0/api/docs/com/google/common/collect/ImmutableList.html#sortedCopyOf-java.util.Comparator-java.lang.Iterable-